cmd/clef/datatypes.md

```diff
@@ -0,0 +1,220 @@
+## UI Client interface
+
+These data types are defined in the channel between clef and the UI
+
+### SignDataRequest
+
+SignDataRequest contains information about a pending request to sign some data.The data to be  signed can be of various types, defined by content-type. Clef has done mostof the work in canonicalizing and making sense of the data, and it's up to the UI to presentthe user with the contents of the `message`
+
+Example:
+```json
+{
+  "content_type": "text/plain",
+  "address": "0xDEADbEeF000000000000000000000000DeaDbeEf",
+  "raw_data": "GUV0aGVyZXVtIFNpZ25lZCBNZXNzYWdlOgoxMWhlbGxvIHdvcmxk",
+  "message": [
+    {
+      "name": "message",
+      "value": "\u0019Ethereum Signed Message:\n11hello world",
+      "type": "text/plain"
+    }
+  ],
+  "hash": "0xd9eba16ed0ecae432b71fe008c98cc872bb4cc214d3220a36f365326cf807d68",
+  "meta": {
+    "remote": "localhost:9999",
+    "local": "localhost:8545",
+    "scheme": "http",
+    "User-Agent": "Firefox 3.2",
+    "Origin": "www.malicious.ru"
+  }
+}
+```
+### SignDataResponse - approve
+
+Reponse to SignDataRequest
+
+Example:
+```json
+{
+  "approved": true,
+  "Password": "apassword"
+}
+```
+### SignDataResponse - deny
+
+Reponse to SignDataRequest
+
+Example:
+```json
+{
+  "approved": false,
+  "Password": ""
+}
+```
+### SignTxRequest
+
+SignTxRequest contains information about a pending request to sign a transaction.Aside from the transaction itself, there is also a call_info-struct. That struct containsmessages of various types, that the user should be informed of.
```

Comment:
SignTxRequest contains information about a pending request to sign a transaction.😱Aside from the transaction itself, there is also a call_info-struct. That struct contains😱messages of various types, that the user should be informed of.

also, I would write it as `call_info`-struct